instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py

```diff
@@ -361,3 +373,35 @@ def wrap_poll(func, instance, tracer, args, kwargs):
         )
 
         return record
+    
+    @staticmethod
+    def wrap_consume(func, instance, tracer, args, kwargs):
+        if instance._current_consume_span:
+            context.detach(instance._current_context_token)
+            instance._current_context_token = None
+            instance._current_consume_span.end()
+            instance._current_consume_span = None
+
+        with tracer.start_as_current_span(
+            "recv", end_on_exit=True, kind=trace.SpanKind.CONSUMER
```

Comment:
What is happening here?

Reply:
It follows the same logic as `poll` method. Let's suppose that your code looks something like this:

```
while(True):
  records = consumer.consume()
  process(records)
```
On the very first iteration of the `while`:

1. It creates a new span `start_as_current_span` and adds all the relevant information (as topic name, links, etc). It saves the span on `_current_consume_span`.
2. Then you process the records you just got from the queue.
3. Finally, you call `consume` again to get more records to process. This piece of code ends the span associated with the records you have just processed (`_current_consume_span`) and repeats the process from 1.